doc/topics/async-system.md

```diff
@@ -17,31 +17,26 @@ Once it's downloaded, we need to process that tile content that we just download
+Unfortunately, there is inevitably at least a little bit of tile loading work that can't be done in a background thread. Game engines, for example, usually have strict rules against creating game objects anywhere other than the main thread. We can't get anything onto the screen without creating game objects, so after the background work is complete, we then need to continue processing this tile in the main thread in order to do the final preparation to render it.
 
 `AsyncSystem` gives us an elegant way to express this kind of sequential process involving a series of asynchronous steps.
 
-## Creating an AsyncSystem {#creating-an-asyncsystem}
+## The AsyncSystem Class
 
-Most applications have a single `AsyncSystem` that is used throughout. It is constructed from an [ITaskProcessor](@ref CesiumAsync::ITaskProcessor) instance, which is a simple interface used to perform some work in a background thread. The simplest possible implementation looks like this:
+An `AsyncSystem` object manages the other objects we use to schedule work and wait for its completion. During initialization, an application [constructs an AsyncSystem object](#creating-an-asyncsystem) and passes it to Cesium Native, which in turn uses it in all operations that might complete asynchronously. `AsyncSystem` instances can be safely and efficiently stored and copied by value; this makes it easy to make them available wherever they're needed, including in lambda captures. On the other hand, holding a reference to an `AsyncSystem` object is very bug-prone in asynchronous code; the lifetime of the reference holder can be quite different from the code that uses it and hard to reason about. It is idiomatic to pass a `const` reference to an `AsyncSystem` object as a parameter to a function, but that reference must be copied to a value if it will be used outside of the lifetime of the function. For example, in the following code, the `asyncSystem` function parameter is copied to a constant value member in the inner lambda:
 
-\snippet{trimleft} ExamplesAsyncSystem.cpp simplest-task-processor
-
-This implementation will work well, but it isn't very efficient because a brand new thread is created for each task. Most applications will implement this interface using a thread pool, task graph, or similar functionality that their application already contains.
-
-The `AsyncSystem` can then be created as follows:
-
-\snippet{trimleft} ExamplesAsyncSystem.cpp create-async-system
-
-`AsyncSystem` instances can be safely and efficiently stored and copied by value. This makes it easy to make them available wherever they're needed, including in lambda captures.
-
-\snippet{trimleft} ExamplesAsyncSystem.cpp capture-by-value
+    Future<ReadJsonResult<Cesium3DTiles::Subtree>> SubtreeFileReader::load(
```

Comment:
Can this example be pulled from the code instead of embedded here? That will increase the chances it continues to be correct over time.